qa/L0_output_fields/test_config_generator.py

```diff
@@ -0,0 +1,96 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import yaml
+
+
+class TestConfigGenerator:
+    """
+    This class contains functions that
+    create configs for various test scenarios.
+    
+    The `setup` function does the work common to all tests
+
+    TO ADD A TEST: Simply add a member function whose name starts
+                    with 'generate'.
+    """
+
+    def __init__(self):
+        test_functions = [
+            self.__getattribute__(name)
+            for name in dir(self)
+            if name.startswith('generate')
+        ]
+
+        for test_function in test_functions:
+            self.setup()
+            test_function()
+
+    def setup(self):
+        pass
+
+    def generate_output_fields_test(self):
+        for i, configuration in enumerate(get_all_configurations()):
+            total_param_server = configuration['total_param_server']
+            total_param_gpu = configuration['total_param_gpu']
+            total_param_inference = configuration['total_param_inference']
+            del configuration['total_param_server']
+            del configuration['total_param_gpu']
+            del configuration['total_param_inference']
+            with open(f'./config-{i}-param-server.txt', 'w') as file:
+                file.write(str(total_param_server))
+            with open(f'./config-{i}-param-gpu.txt', 'w') as file:
+                file.write(str(total_param_gpu))
+            with open(f'./config-{i}-param-inference.txt', 'w') as file:
+                file.write(str(total_param_inference))
+            with open(f'./config-{i}.yml', 'w') as file:
+                yaml.dump(configuration, file)
+
+
+def _get_sweep_configs():
+
+    sweep_configs = []
+    model_config = {
+        'analysis_models': ['vgg19_libtorch'],
+        'server_output_fields': [
+            'model_name', 'gpu_id', 'gpu_used_memory', 'gpu_utilization'
+        ],
+        'gpu_output_fields': [
+            'model_name', 'satisfies_constraints', 'gpu_used_memory',
+            'gpu_utilization', 'gpu_power_usage'
+        ],
+        'inference_output_fields': [
+            'model_name', 'batch_size', 'concurrency', 'model_config_path',
+            'perf_throughput', 'perf_latency', 'cpu_used_ram'
+        ]
+    }
+
+    model_config['total_param_server'] = len(
+        model_config['server_output_fields'])
+    model_config['total_param_gpu'] = len(model_config['gpu_output_fields'])
+    model_config['total_param_inference'] = len(
+        model_config['inference_output_fields'])
+    sweep_configs.append(model_config)
+    return sweep_configs
+
+
+def get_all_configurations():
```

Comment:
You can delete this and just use `_get_sweep_configs` from `generate_output_fields_test`

Reply:
Revert to use the original `config_generator.py`